src/main/java/Duke.java

```diff
@@ -1,204 +1,58 @@
-import java.util.ArrayList;
-import java.util.Scanner;
-import java.nio.file.Paths;
-import java.nio.file.Files;
-import java.io.FileWriter;
 import java.io.IOException;
-import java.io.File;
-import java.io.FileNotFoundException;
-
+import java.util.Scanner;
 
 public class Duke {
 
-    private Scanner input;
-    private ArrayList<Task> list;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
 
-    Duke(Scanner input, ArrayList<Task> list) {
-        this.input = input;
-        this.list = list;
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        tasks = new TaskList(storage.load());
     }
 
-    void commandHandler() {
-        while (input.hasNextLine()) {
-            String command = input.nextLine();
-            System.out.println("___________________________________________________");
-            try {
-                if (command.equals("bye")) {
-                    this.saveTasks();
+    public void run() {
+        try {
+            ui.intro();
+            Scanner input = new Scanner(System.in);
+            loop:
+            while (input.hasNextLine()) {
+                String line = input.nextLine();
+                Command command = ui.parseCommand(line);
+                ui.lineBreak();
+                switch (command.name) {
```

Comment:
good use of switch case to list all possible outcome, adding a default may be better